internal/resources/slo/resource_slo.go

```diff
@@ -0,0 +1,477 @@
+package slo
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/grafana/terraform-provider-grafana/internal/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceSlo() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+Resource manages Grafana SLOs. 
+
+* [Official documentation](https://grafana.com/docs/grafana-cloud/slo/)
+* [API documentation](https://grafana.com/docs/grafana-cloud/slo/api/)
+		`,
+		CreateContext: resourceSloCreate,
+		ReadContext:   resourceSloRead,
+		UpdateContext: resourceSloUpdate,
+		DeleteContext: resourceSloDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Name should be a short description of your indicator. Consider names like "API Availability"`,
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Description is a free-text field that can provide more context to an SLO.`,
+			},
+			"query": &schema.Schema{
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Required:    true,
+				Description: `Query describes the indicator that will be measured against the objective. Freeform Query types are currently supported.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:        schema.TypeString,
+							Description: `Query type must be one of: "freeform", "query", "ratio", or "threshold"`,
```

Comment:
Proper validation here?
```suggestion
							Description: `Query type must be one of: "freeform", "query", "ratio", or "threshold"`,
							ValidateFunc: validation.StringInSlice([]string{"freeform", "query", "ratio", "threshold"}, false),
```